### PR TITLE
Add couchdb 2.1.0 docker image

### PR DESCRIPTION
**Pull request by nicolai86**
## Overview

this PR adjusts the `2.0.0` setup to build `2.1.0`

## Testing recommendations

```
$ cd 2.1.0
$ docker build -t couchdb:2.1.0 .
```

### CODE REVIEW

File: 2.1.0/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+
+COPY ./docker-entrypoint.sh /
+
+# Setup directories and permissions
+RUN chmod +x /docker-entrypoint.sh \
+ && mkdir /opt/couchdb/data \
+ && chown -R couchdb:couchdb /opt/couchdb/
+
+WORKDIR /opt/couchdb
+EXPOSE 5984 4369 9100
```
Comment (wohali) [inline 2.1.0/Dockerfile]: Add 5986 to this list please

File: 2.1.0/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+    libmozjs185-dev \
+    make \
+  ' \
+ && apt-get update -y -qq && apt-get install -y --no-install-recommends $buildDeps \
+ && curl -s https://deb.nodesource.com/gpgkey/nodesource.gpg.key | apt-key add - \
+ && echo 'deb https://deb.nodesource.com/node_4.x jessie main' > /etc/apt/sources.list.d/nodesource.list \
+ && echo 'deb-src https://deb.nodesource.com/node_4.x jessie main' >> /etc/apt/sources.list.d/nodesource.list \
+ && apt-get update -y -qq \
+ && apt-get install -y nodejs \
+ && npm install -g grunt-cli \
```
Comment (wohali) [inline 2.1.0/Dockerfile]: This entire section, LL77-82, should be removable. We pre-build Fauxton for you in the release tarball, so node/npm/grunt should not be necessary. Please remove and test the build?

File: 2.1.0/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+ && gpg --verify couchdb.tar.gz.asc \
+ && tar -xzf couchdb.tar.gz -C couchdb --strip-components=1 \
+ && cd couchdb \
+ # Build the release and install into /opt
+ && ./configure --disable-docs \
+ && make release \
+ && mv /usr/src/couchdb/rel/couchdb /opt/ \
+ # Cleanup build detritus
+ && apt-get purge -y --auto-remove $buildDeps \
+ && rm -rf /var/lib/apt/lists/* /usr/lib/node_modules /usr/src/couchdb*
```
Comment (wohali) [inline 2.1.0/Dockerfile]: If you remove LL77-82 you should be able to remove /usr/lib/node_modules from this `rm` command.

**Comment by nicolai86** [comment]
Thank you so much for the review, @wohali . I'll push an update soon.

**Comment by nicolai86** [comment]
@wohali please take another look :) works w/o node.

**Comment by wohali** [comment]
I'd like to hear from @calonso that this works for him before I merge, but it looks good in its current state.

**Comment by calonso** [comment]
So I've built @nicolai86 's fork and upgraded a test 2.0.0 cluster to 2.1.0 and everything worked fine. Just a couple of things to note. During the upgrade process (when both versions were coexisting):

* Running replications appear twice on Fauxton (that happened when the owner of the replication was taken down and another node took it over).
* Once the owner comes back, the replication is restarted (data is not transferred, just the `_revs_diff` procedure)
* A `rexi_server error:function clause` error appears once per shard of every db. 

```
[notice] 2017-08-23T10:50:33.905116Z couchdb@couch-0 <0.358.0> -------- Maybe starting 'f86e949983791ba2fc901b8d96000576' as I'm the owner
[notice] 2017-08-23T10:50:33.925678Z couchdb@couch-0 <0.358.0> -------- updating replicator ddoc
[error] 2017-08-23T10:50:34.046089Z couchdb@couch-0n <0.25171.11> -------- rexi_server error:function_clause [{fabric_rpc,all_docs,[<<"shards/00000000-1fffffff/_replicator.1503484582">>,[{user_ctx,{user_ctx,null,[<<"_admin">>],undefined}}],{mrargs,undefined,undefined,undefined,undefined,undefined,undefined,undefined,undefined,fwd,268435456,0,0,undefined,false,true,false,true,true,[],false,undefined,undefined,true,[{namespace,<<"_design">>}]}],[{file,"src/fabric_rpc.erl"},{line,95}]},{rexi_server,init_p,3,[{file,"src/rexi_server.erl"},{line,139}]}]
...
```

But in the end, as I said before, everything seemed to be working fine except for the fact that the running replications are restarted.

**Comment by nickva** [comment]
* The ownership function (which node each replication task should run on) has changed so it is expected for a short period of time when the cluster both version to have some replications running on more than one node. So that  part is expected.

 * Replication would be restarted on eventually and should be running on only one node. It may take up to a minute after the upgrade finished for that to happen.

* Not sure about the function clause error. Does it persist? Or just saw a single instance of it. That might be coming from Fauxton querying the _replicator db. Replicator application uses a changes feed instead of all_docs (noticed all_docs in there).

**Comment by calonso** [comment]
Hi @nickva 

Thanks for your explanations. The error doesn't persist. It appears once per shard per db.

**Comment by nickva** [comment]
@calonso ok, cool, I realized it as well by rereading your comment

**Comment by nickva** [comment]
Discussing in couchdb-dev channel @rnewson has the right idea that it might that the size of the mrargs record has changed and indeed it has:

https://github.com/apache/couchdb/commit/459a548f269b00bf2d2ace0170c583e72fde4cb9

So that's where the function clause error is coming from.

**Comment by wohali** [comment]
I'll merge this. We will then track the image deployment over in #23.

Thanks for your contribution to Apache CouchDB!

**Pull request merged**